app/uk/gov/hmrc/economiccrimelevyaccount/models/FinancialData.scala

```diff
@@ -57,6 +57,12 @@ case class FinancialData(totalisation: Option[Totalisation], documentDetails: Op
       })
       .flatMap(_.sortBy(_.postingDate).headOption)
 
+  def getFinancialObligation(chargeReference: String): Option[DocumentDetails] =
+    documentDetails
+      .map(_.collect {
```

Comment:
```
    documentDetails.flatMap(_.collectFirst{
      case docDetails if docDetails.chargeReferenceNumber.contains(chargeReference) => docDetails
    })
```

Can use collectFirst to simplify it a little

Reply:
Changed, thanks.